.travis.yml

```diff
@@ -31,44 +38,94 @@ jobs:
   include:
     - php: nightly
       env: COMPOSER_FLAGS="--ignore-platform-reqs"
+
     - stage: test with coverage
       php: 7.1
       before_install:
         - stty cols 130
         - echo 'opcache.enable=1' >> ~/.phpenv/versions/$(phpenv version-name)/etc/php.ini
         - echo 'opcache.enable_cli=1' >> ~/.phpenv/versions/$(phpenv version-name)/etc/php.ini
         - travis_retry composer self-update
-      script: vendor/bin/phpunit -c phpunit.xml.dist --colors --columns 121 --verbose --coverage-text --coverage-clover=coverage.clover
+      script: vendor/bin/phpunit -c phpunit.xml.dist --colors --columns 117 --verbose --coverage-text --coverage-clover=coverage.clover
       after_success:
         - wget https://scrutinizer-ci.com/ocular.phar && php ocular.phar code-coverage:upload --format=php-clover coverage.clover
-        - bash <(curl -s https://codecov.io/bash) -f coverage.clover
+        - bash <(curl -s https://codecov.io/bash) -f coverage.clover -F phpunit
+
+    - stage: composer validate
+      php: 7.1
+      before_install:
+        - stty cols 130
+        - phpenv config-rm xdebug.ini || echo "xdebug not available"
+        - echo 'opcache.enable=1' >> ~/.phpenv/versions/$(phpenv version-name)/etc/php.ini
+        - echo 'opcache.enable_cli=1' >> ~/.phpenv/versions/$(phpenv version-name)/etc/php.ini
+      install: skip
+      script: composer validate --strict
+
     - stage: check coding style
       php: 7.1
       script: php -n -d memory_limit=768M vendor/bin/php-cs-fixer fix --dry-run -vv
+
     - stage: static code analysis
       php: 7.1
       script: vendor/bin/phpstan analyse -l 7 -c phpstan.neon --autoload-file=vendor/autoload.php --memory-limit=768M --no-progress src tests/UserAgentsTest tests/BrowscapTest tests/fixtures
+
     - stage: check for duplicate integration tests
       php: 7.1
       script: bin/browscap -v check-duplicate-tests
+
+    - stage: integration tests
+      php: 7.1
+      env:
+        - parser="browscap-php 4.x@dev"
+        - TEST_SET="full"
+      install:
+        - travis_retry composer update --optimize-autoloader --prefer-dist --prefer-stable --no-progress --no-interaction -vv $COMPOSER_FLAGS
+        - travis_retry composer require --update-with-dependencies --prefer-dist --no-suggest browscap/browscap-php:dev-master
+      script: php -n vendor/bin/phpunit -c tests/phpunit-integration.xml.dist --no-coverage --colors --columns 121 --verbose tests/UserAgentsTest/Full4Test.php
+      after_success: bash <(curl -s https://codecov.io/bash) -f coverage-full4.json -F full4
```

Comment:
I don't think it's important to run coverage for 3.x and 4.x runs, we should just pick one.

Reply:
changed